src/symboltable.c

```diff
@@ -375,6 +375,13 @@ void vmp_scan_profile(int fileno, int dump_nat_sym, void *all_code_uids)
     khash_t(ptr) * nat_syms = kh_init(ptr);
     khiter_t it;
 
+#ifndef RPYTHON_VMPROF
+    PyObject *all_done_uids = NULL;
+    if (all_code_uids != NULL) {
+        all_done_uids = PySet_New(NULL);
```

Comment:
in this function we cant use CPython API. It will not work on PyPy. I have used khash for the native symbols already (to avoid duplication). We can reuse khash for that.